aws_lambda_powertools/utilities/idempotency/persistence/base.py

```diff
@@ -0,0 +1,441 @@
+"""
+Persistence layers supporting idempotency
+"""
+
+import datetime
+import hashlib
+import json
+import logging
+from abc import ABC, abstractmethod
+from typing import Any, Dict
+
+import jmespath
+
+from aws_lambda_powertools.shared.cache_dict import LRUDict
+from aws_lambda_powertools.utilities.idempotency.exceptions import (
+    IdempotencyInvalidStatusError,
+    IdempotencyItemAlreadyExistsError,
+    IdempotencyValidationError,
+)
+
+logger = logging.getLogger(__name__)
+
+STATUS_CONSTANTS = {
+    "NOTEXISTING": "DOESNOTEXIST",
```

Comment:
__Nitpick__: Is there a reason why the names here are different?

Reply:
Yes, but not a good one! I removed a couple of unused constants now.